### DISCUSSION TITLE
How to add attributes for all (custom and autogenerated) metrics which are auto generated by Java agent and exported by Prometheus exporter.

### DISCUSSION THREAD

**Discussion by lokeshmavale** [open-telemetry/opentelemetry-java#5163] [Q&A] [answered]
Hi Team,
I have requirement to send metrics in OTLP & Prometheus format. While sending metric I want to enrich the attributes for all the metrics (custom and autogenerated).

With Auto Configure SPI, I was able to override metrics using `autoConfiguration.addMetricExporterCustomizer(this::configureSdkMetricExporterProvider);`

I have observed that, this exporter only has access to OTLP (push based) exporter.
How can I override Prometheus exporter without creating all new Prometheus server and registering it with auto agent using `autoConfiguration.addMeterProviderCustomizer().` ?

Do we support pull based model exporter override via auto configuration?

Thanks,
Lokesh

**Comment by mateuszrzeszutek** [comment, marked as answer]
Hey @lokeshmavale ,

AFAIK currently there's no feature like that in the SDK. You can use the collector's [attributes processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/attributesprocessor) to modify the attributes though.

**Reply by lokeshmavale** [reply]
Thanks for quick response @mateuszrzeszutek, as collector's codebase is externally owned and maintained, I think that's the only above option.
